Modules/Core/SpatialObjects/include/itkSpatialObject.h

```diff
@@ -522,8 +522,11 @@ class ITK_TEMPLATE_EXPORT SpatialObject:
   /** Compute bounding box for the object in world space */
   virtual void ComputeMyBoundingBox();
 
-  /** Constructor. */
-  SpatialObject();
+  /** Default constructor. Ensures that its bounding boxes are empty (all
+  * bounds zero-valued), its list of children is empty, and its transform
+  * objects identical to the identity transform, initially.
+  */
+  SpatialObject() = default;
```

Comment:
For libraries with compiles objects it is best practice to keep the implementation in the compiled cxx files. Here is a little example of the problems that can arise with shared libraries when the descrutor is implicitly instantiated:
https://github.com/blowekamp/PrivateDynamicCastTest/pull/1

The destructor should be explicitly instantiated and compiled in the library.